python-package/xlearn/sklearn.py

```diff
@@ -0,0 +1,185 @@
+import os
+import shutil
+import tempfile
+import numpy as np
+
+from .xlearn import create_linear, create_fm, create_ffm
+from sklearn.base import BaseEstimator
+from sklearn.datasets import dump_svmlight_file
+
+class BaseXLearnModel(BaseEstimator):
+    """ Implementation of Scikit-learn interface for xlearn models.
+    """
+    def __init__(self, model_type='fm', task='binary', metric='auc',
+                        lr=0.2, k =4, reg_lambda=0.1, init=0.1, fold=1, epoch=5,
+                        opt='sgd', nthread=4, alpha=1, beta=1, lambda_1=1, lambda_2=1,
+                        **kwargs):
```

Comment:
How to set the other hyper-parameters like early-stop, norm, or lock-free learning ?

Reply:
I will expose early-stop, norm and lock-free learning in the fit method, as these parameters are only related to training.